content/blog/2019/04/2019-04-15-templating-engine.adoc

```diff
@@ -0,0 +1,285 @@
+---
+:layout: post
+:title: "Introducing the Jenkins Templating Engine!"
+:tags:
+- general
+- pipeline
+- plugin
+- pipeline-authoring
+:author: Steven Terrana
+:sig: pipeline-authoring
+---
+
+[NOTE]
+====
+This is a guest post by link:https://github.com/steven-terrana[Steven Terrana], a Lead Technologist at
+link:https://boozallen.com[Booz Allen Hamilton] and principal engineer working on the plugin:templating-engine[Templating Engine Plugin]. 
+He participates in the link:../../../../../sigs/pipeline-authoring[Pipeline Authoring  Special Interest Group].
+====
+
+
+Implementing DevSecOps practices at the enterprise scale is challenging. With multiple programming languages, automated testing frameworks, and security compliance tools being used by different applications within your organization, it becomes difficult to build and maintain pipelines for each team. 
+
+Most pipelines are going to follow the same generic workflow regardless of which specific tech stack is employed by an application.  The plugin:templating-engine[Templating Engine Plugin] (abbreviated as JTE for Jenkins Templating Engine) allows you to capture this efficiency by creating tool-agnostic, templated workflows to be reused by every team.  
+
+As technology consultants with clients in both the public and private sectors, at Booz Allen we found ourselves building DevSecOps pipelines from scratch for every new project.  Through developing the Jenkins Templating Engine, we’ve seen pipeline development decrease from months to days now that we can reuse tool integrations while bringing a new level of governance to Jenkins pipelines. 
+
+
+= Pipeline Templating 
+
+Organizations benefit from letting application developers focus on what they do best: building applications. Supporting this means building a centralized DevOps team responsible for maintaining platform infrastructure and creating CI/CD pipelines utilized by development teams.  
+
+With the rise of microservice-based architectures, a centralized DevOps teams can support many different development teams simultaneously; all of whom may be leveraging different programming languages and automating testing tools.  
```

Comment:
"a centralized DevOps team"?

Also, s/automating/automated ?

Reply:
Thank you for the catch!  Updated "automating testing" to "automated testing". 

For the "centralized DevOps team" i'm referring to a pattern often seen where a centralized DevOps team builds and maintains common services that application development teams utilize.  For instance, maintaining a multi-tenant Jenkins server or SonarQube instance that developers will then utilize when developing their apps. 

I can clarify this language or if that explanation is sufficient we can leave it as is.